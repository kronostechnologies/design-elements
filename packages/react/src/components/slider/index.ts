export { Slider } from './slider';
export type { SliderProps } from './slider';
