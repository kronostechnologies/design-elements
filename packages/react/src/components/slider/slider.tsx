import { FunctionComponent, ReactElement } from 'react';
import { Slider as BaseSlider, SliderValueLabelSlotProps, sliderClasses } from '@mui/base/Slider';
import styled from 'styled-components';
import { TooltipProps } from '../tooltip/tooltip';
import { useId } from '../../hooks/use-id';
import { FieldContainer } from '../field-container/field-container';

export const ValueTooltip = styled.div`
    background-color: ${({ theme }) => theme.component['tooltip-popper-container-default-background-color']};
    border: 1px solid ${({ theme }) => theme.component['tooltip-popper-container-border-color']};
    border-radius: var(--border-radius);
    box-shadow: 0 10px 20px 0 rgb(0 0 0 / 19%);
    box-sizing: border-box;
    color: ${({ theme }) => theme.component['tooltip-popper-container-text-color']};
    font-size: 0.75rem;
    padding: var(--spacing-half) var(--spacing-1x);
    position: absolute;
    top: 2.5rem;
    transform: scale(0);
    transform-origin: 50% -0.49rem;
    transition: transform 0.12s;

    &::before {
        border-color: transparent transparent ${({ theme }) => theme.component['tooltip-popper-container-default-background-color']} transparent;
        border-style: solid;
        border-width: 0 0.5rem 0.5rem;
        content: '';
        height: 0;
        left: 50%;
        position: absolute;
        top: -0.49rem;
        transform: translateX(-50%);
        width: 0;
    }
`;

const Labels = styled.div`
    color: ${({ theme }) => theme.component['slider-rail-label-text-color']};
    display: flex;
    font-size: 0.75rem;
    justify-content: space-between;
    margin-top: 2px;
`;

const SliderContainer = styled.div`
    padding: 0 12px;
`;

const StyledSlider = styled(BaseSlider)`
    align-items: center;
    cursor: pointer;
    display: flex;
    height: 8px;
    padding: 8px 0;
    position: relative;
    -webkit-tap-highlight-color: transparent;
    touch-action: none;
    width: 100%;

    & .${sliderClasses.rail} {
        background-color: ${({ theme }) => theme.component['slider-rail-background-color']};
        border-radius: 99px;
        display: block;
        height: 8px;
        margin-left: -12px;
        padding: 0 12px;
        position: absolute;
        width: 100%;
    }

    & .${sliderClasses.track} {
        background-color: ${({ theme }) => theme.component['slider-track-background-color']};
        border-radius: 99px;
        display: block;
        height: 8px;
        margin-left: -12px;
        padding: 0 12px;
        position: absolute;
    }

    & .${sliderClasses.thumb} {
        border-radius: 50%;
        display: flex;
        height: 32px;
        justify-content: center;
        margin-left: -16px;
        position: absolute;
        width: 32px;

        &::before {
            background-color: ${({ theme }) => theme.component['slider-thumb-background-color']};
            border: 2px solid ${({ theme }) => theme.component['slider-thumb-border-color']};
            border-radius: 50%;
            box-sizing: border-box;
            content: '';
            display: block;
            height: 24px;
            left: 50%;
            outline: 0;
            position: absolute;
            top: 50%;
            transform: translate(-50%, -50%);
            transform-origin: center;
            transition-duration: 120ms;
            transition-property: box-shadow, width, height;
            width: 24px;
        }
        
        &.${sliderClasses.active},
        &:hover {
            &::before {
                height: 32px;
                width: 32px;
            }
        }

        &.${sliderClasses.focusVisible} {
            &::before {
                box-shadow: inset 0 0 0 2px ${({ theme }) => theme.component['focus-inside-border-color']};
            }
        }
        
        &.${sliderClasses.active} {
            &::before {
                box-shadow: inset 0 0 0 2px ${({ theme }) => theme.component['slider-thumb-border-color']};
            }
        }

        &.${sliderClasses.focusVisible},
        &:hover {
            ${ValueTooltip} {
                transform: scale(1);
            }
        }
    }

    & .${sliderClasses.mark} {
        background-color: ${({ theme }) => theme.component['slider-rail-mark-background-color']};
        border-radius: 99%;
        height: 2px;
        position: absolute;
        transform: translateX(-50%);
        width: 2px;
    }

    & .${sliderClasses.markActive} {
        background-color: ${({ theme }) => theme.component['slider-track-mark-background-color']};
    }

    &.${sliderClasses.disabled} {
        cursor: default;
        pointer-events: none;
        
        .${sliderClasses.rail} {
            background-color: ${({ theme }) => theme.component['slider-rail-disabled-background-color']};
        }

        .${sliderClasses.track} {
            background-color: ${({ theme }) => theme.component['slider-track-disabled-background-color']};
        }

        .${sliderClasses.thumb} {
            &::before {
                border-color: ${({ theme }) => theme.component['slider-thumb-disabled-border-color']};
            }
        }

        .${sliderClasses.mark} {
            background-color: ${({ theme }) => theme.component['slider-rail-mark-disabled-background-color']};
        }

        & .${sliderClasses.markActive} {
            background-color: ${({ theme }) => theme.component['slider-track-mark-disabled-background-color']};
        }
    }
`;

const SliderValueTooltip: FunctionComponent<SliderValueLabelSlotProps> = ({ children }) => (
    <ValueTooltip>
        {children}
    </ValueTooltip>
);

// Widen the literal value to be number (ex: type '20' -> type 'number')
type InferValueType<T> = T extends number ? number : number[];

interface SliderProps<TValue extends number | number[]> {
    className?: string;
    id?: string;
    max: number;
    min: number;
    value?: TValue;
    disabled?: boolean;
    defaultValue?: TValue;
    hint?: string;
    label?: string;
    noMargin?: boolean;
    tooltip?: TooltipProps;
    invalid?: boolean;
    validationErrorMessage?: string;
    step?: number;
    onChange?: (event: Event, value: InferValueType<TValue>) => void;
}

// eslint-disable-next-line comma-spacing
export const Slider = <TValue extends number | number[]>({
    className,
    id: providedId,
    max,
    min,
    value,
    defaultValue,
    disabled,
    hint,
    label,
    noMargin,
    tooltip,
    invalid,
    validationErrorMessage,
    step,
    onChange,
}: SliderProps<TValue>): ReactElement => {
    const id = useId(providedId);

    return (
        <FieldContainer
            className={className}
            fieldId={id}
            hint={hint}
            label={label}
            tooltip={tooltip}
            noMargin={noMargin}
            valid={!invalid}
            noInvalidFieldIcon={!validationErrorMessage}
            validationErrorMessage={validationErrorMessage ?? ''}
        >
            <SliderContainer>
                <StyledSlider
                    data-testid="slider"
                    min={min}
                    max={max}
                    value={value}
                    defaultValue={defaultValue}
                    disabled={disabled}
                    step={step}
                    marks={step ? true : undefined}
                    slots={{ valueLabel: SliderValueTooltip }}
                    slotProps={{
                        input: { id },
                    }}
                    onChange={onChange}

                />
                {/* <BaseSlider slotProps={{ valueLabel: }}> */}
            </SliderContainer>
            <Labels>
                <span>{min}</span>
                <span>{max}</span>
            </Labels>
        </FieldContainer>
    );
};
