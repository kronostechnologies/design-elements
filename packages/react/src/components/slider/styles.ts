import { Slider as BaseSlider, sliderClasses } from '@mui/base/Slider';
import styled from 'styled-components';
import { ValueTooltipContainer } from './value-tooltip';

export const Labels = styled.div`
    color: ${({ theme }) => theme.component['slider-track-empty-label-text-color']};
    display: flex;
    font-size: 0.75rem;
    justify-content: space-between;
    margin-top: 2px;
`;

export const SliderContainer = styled.div`
    padding: 0 12px;
`;

export const StyledSlider = styled(BaseSlider)`
    align-items: center;
    cursor: pointer;
    display: flex;
    height: 8px;
    padding: 8px 0;
    position: relative;
    -webkit-tap-highlight-color: transparent;
    touch-action: none;
    width: 100%;

    & .${sliderClasses.rail} {
        background-color: ${({ theme }) => theme.component['slider-track-empty-background-color']};
        border-radius: 99px;
        display: block;
        height: 8px;
        margin-left: -12px;
        padding: 0 12px;
        position: absolute;
        width: 100%;
    }

    & .${sliderClasses.track} {
        background-color: ${({ theme }) => theme.component['slider-track-filled-background-color']};
        border-radius: 99px;
        display: block;
        height: 8px;
        margin-left: -12px;
        padding: 0 12px;
        position: absolute;
    }

    & .${sliderClasses.thumb} {
        border-radius: 50%;
        display: flex;
        height: 32px;
        justify-content: center;
        margin-left: -16px;
        position: absolute;
        width: 32px;

        &::before {
            background-color: ${({ theme }) => theme.component['slider-thumb-background-color']};
            border: 2px solid ${({ theme }) => theme.component['slider-thumb-border-color']};
            border-radius: 50%;
            box-sizing: border-box;
            content: '';
            display: block;
            height: 24px;
            left: 50%;
            outline: 0;
            position: absolute;
            top: 50%;
            transform: translate(-50%, -50%);
            transform-origin: center;
            transition-duration: 120ms;
            transition-property: box-shadow, width, height;
            width: 24px;
        }
        
        &.${sliderClasses.active},
        &:hover {
            &::before {
                height: 32px;
                width: 32px;
            }
        }

        &.${sliderClasses.focusVisible} {
            &::before {
                box-shadow: inset 0 0 0 2px ${({ theme }) => theme.component['focus-inside-border-color']};
            }
        }
        
        &.${sliderClasses.active} {
            &::before {
                box-shadow: inset 0 0 0 2px ${({ theme }) => theme.component['slider-thumb-border-color']};
            }
        }

        &.${sliderClasses.focusVisible},
        &:hover {
            ${ValueTooltipContainer} {
                transform: scale(1);
            }
        }
    }

    & .${sliderClasses.mark} {
        background-color: ${({ theme }) => theme.component['slider-track-empty-mark-background-color']};
        border-radius: 99%;
        height: 2px;
        position: absolute;
        transform: translateX(-50%);
        width: 2px;
    }

    & .${sliderClasses.markActive} {
        background-color: ${({ theme }) => theme.component['slider-track-filled-mark-background-color']};
    }

    &.${sliderClasses.disabled} {
        cursor: default;
        pointer-events: none;
        
        .${sliderClasses.rail} {
            background-color: ${({ theme }) => theme.component['slider-track-empty-disabled-background-color']};
        }

        .${sliderClasses.track} {
            background-color: ${({ theme }) => theme.component['slider-track-filled-disabled-background-color']};
        }

        .${sliderClasses.thumb} {
            &::before {
                border-color: ${({ theme }) => theme.component['slider-thumb-disabled-border-color']};
            }
        }

        .${sliderClasses.mark} {
            background-color: ${({ theme }) => theme.component['slider-track-empty-mark-disabled-background-color']};
        }

        & .${sliderClasses.markActive} {
            background-color: ${({ theme }) => theme.component['slider-track-filled-mark-disabled-background-color']};
        }
    }
`;
