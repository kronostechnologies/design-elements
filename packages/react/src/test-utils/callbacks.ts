export function doNothing(): void {
    // do nothing
}
