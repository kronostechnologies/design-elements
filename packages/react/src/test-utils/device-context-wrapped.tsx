import React, { ReactElement } from 'react';
import { DeviceContextProvider, DeviceType } from '../components/device-context-provider/device-context-provider';

export function DeviceContextWrapped(children: ReactElement, device?: DeviceType): ReactElement {
    return (
        <DeviceContextProvider staticDevice={device}>
            {children}
        </DeviceContextProvider>
    );
}
