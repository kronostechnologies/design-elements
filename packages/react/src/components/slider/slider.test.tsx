import { renderWithProviders } from '../../test-utils/renderer';
import { Slider } from './slider';

describe('Slider', () => {
    test('matches snapshot (single value)', () => {
        const wrapper = renderWithProviders(<Slider label="Label" max={100} min={0} value={20} />);

        expect(wrapper).toMatchSnapshot();
    });

    test('matches snapshot (range values)', () => {
        const wrapper = renderWithProviders(<Slider label="Label" max={100} min={0} value={[10, 30]} />);

        expect(wrapper).toMatchSnapshot();
    });

    test('matches snapshot (with step)', () => {
        const wrapper = renderWithProviders(<Slider label="Label" max={100} min={0} value={30} step={10} />);

        expect(wrapper).toMatchSnapshot();
    });
});
