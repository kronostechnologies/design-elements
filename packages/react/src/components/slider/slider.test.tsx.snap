// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider matches snapshot (range values) 1`] = `
.c2 {
  color: #1B1C1E;
  display: block;
  font-size: 0.75rem;
  font-weight: var(--font-normal);
  -webkit-letter-spacing: 0.02rem;
  -moz-letter-spacing: 0.02rem;
  -ms-letter-spacing: 0.02rem;
  letter-spacing: 0.02rem;
  line-height: 1.25rem;
  margin: 0;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

input + .c1 {
  margin-left: var(--spacing-half);
}

.c0 {
  margin: 0 0 var(--spacing-3x);
}

.c0 input,
.c0 select,
.c0 textarea {
  border-color: #60666E;
}

.c0 > :nth-child(1) {
  margin-bottom: var(--spacing-half);
}

.c6 {
  background-color: #60666E;
  border: 1px solid #FFFFFF;
  border-radius: var(--border-radius);
  box-shadow: 0 10px 20px 0 rgb(0 0 0 / 19%);
  box-sizing: border-box;
  color: #FFFFFF;
  font-size: 0.75rem;
  padding: var(--spacing-half) var(--spacing-1x);
  position: absolute;
  top: 2.5rem;
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  -webkit-transform-origin: 50% -0.49rem;
  -ms-transform-origin: 50% -0.49rem;
  transform-origin: 50% -0.49rem;
  -webkit-transition: -webkit-transform 0.12s;
  -webkit-transition: transform 0.12s;
  transition: transform 0.12s;
}

.c6::before {
  border-color: transparent transparent #60666E transparent;
  border-style: solid;
  border-width: 0 0.5rem 0.5rem;
  content: '';
  height: 0;
  left: 50%;
  position: absolute;
  top: -0.49rem;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  width: 0;
}

.c7 {
  color: #60666E;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 0.75rem;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-top: 2px;
}

.c3 {
  padding: 0 12px;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 8px;
  padding: 8px 0;
  position: relative;
  -webkit-tap-highlight-color: transparent;
  touch-action: none;
  width: 100%;
}

.c4 .base-Slider-rail {
  background-color: #DBDEE1;
  border-radius: 99px;
  display: block;
  height: 8px;
  margin-left: -12px;
  padding: 0 12px;
  position: absolute;
  width: 100%;
}

.c4 .base-Slider-track {
  background-color: #006296;
  border-radius: 99px;
  display: block;
  height: 8px;
  margin-left: -12px;
  padding: 0 12px;
  position: absolute;
}

.c4 .base-Slider-thumb {
  border-radius: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 32px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-left: -16px;
  position: absolute;
  width: 32px;
}

.c4 .base-Slider-thumb::before {
  background-color: #FFFFFF;
  border: 2px solid #006296;
  border-radius: 50%;
  box-sizing: border-box;
  content: '';
  display: block;
  height: 24px;
  left: 50%;
  outline: 0;
  position: absolute;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  -webkit-transition-duration: 120ms;
  transition-duration: 120ms;
  -webkit-transition-property: box-shadow,width,height;
  transition-property: box-shadow,width,height;
  width: 24px;
}

.c4 .base-Slider-thumb.base--active::before,
.c4 .base-Slider-thumb:hover::before {
  height: 32px;
  width: 32px;
}

.c4 .base-Slider-thumb.base--focusVisible::before {
  box-shadow: inset 0 0 0 2px #84C6EA;
}

.c4 .base-Slider-thumb.base--active::before {
  box-shadow: inset 0 0 0 2px #006296;
}

.c4 .base-Slider-thumb.base--focusVisible .c5,
.c4 .base-Slider-thumb:hover .c5 {
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.c4 .base-Slider-mark {
  background-color: #60666E;
  border-radius: 99%;
  height: 2px;
  position: absolute;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  width: 2px;
}

.c4 .base-Slider-markActive {
  background-color: #FFFFFF;
}

.c4.base--disabled {
  cursor: default;
  pointer-events: none;
}

.c4.base--disabled .base-Slider-rail {
  background-color: #F1F2F2;
}

.c4.base--disabled .base-Slider-track {
  background-color: #B7BBC2;
}

.c4.base--disabled .base-Slider-thumb::before {
  border-color: #B7BBC2;
}

.c4.base--disabled .base-Slider-mark {
  background-color: #B7BBC2;
}

.c4.base--disabled .base-Slider-markActive {
  background-color: #FFFFFF;
}

<div
  class="c0"
>
  <label
    class="c1 c2"
    for="uuid1"
    id="uuid1_label"
  >
    Label
  </label>
  <div
    class="c3"
  >
    <span
      class="base-Slider-root c4"
      data-testid="slider"
    >
      <span
        class="base-Slider-rail"
      />
      <span
        class="base-Slider-track"
        style="left:10%;width:20%"
      />
      <span
        class="base-Slider-thumb"
        data-index="0"
        style="left:10%"
      >
        <input
          aria-label="Label range"
          aria-orientation="horizontal"
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="10"
          data-index="0"
          max="100"
          min="0"
          step="1"
          style="border:0;clip:rect(0 0 0 0);height:100%;margin:-1px;overflow:hidden;padding:0;position:absolute;white-space:nowrap;width:100%;direction:ltr"
          type="range"
          value="10"
        />
        <div
          class="c5 c6"
        >
          10
        </div>
      </span>
      <span
        class="base-Slider-thumb"
        data-index="1"
        style="left:30%"
      >
        <input
          aria-label="Label range"
          aria-orientation="horizontal"
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="30"
          data-index="1"
          max="100"
          min="0"
          step="1"
          style="border:0;clip:rect(0 0 0 0);height:100%;margin:-1px;overflow:hidden;padding:0;position:absolute;white-space:nowrap;width:100%;direction:ltr"
          type="range"
          value="30"
        />
        <div
          class="c5 c6"
        >
          30
        </div>
      </span>
    </span>
  </div>
  <div
    class="c7"
  >
    <span>
      0
    </span>
    <span>
      100
    </span>
  </div>
</div>
`;

exports[`Slider matches snapshot (single value) 1`] = `
.c2 {
  color: #1B1C1E;
  display: block;
  font-size: 0.75rem;
  font-weight: var(--font-normal);
  -webkit-letter-spacing: 0.02rem;
  -moz-letter-spacing: 0.02rem;
  -ms-letter-spacing: 0.02rem;
  letter-spacing: 0.02rem;
  line-height: 1.25rem;
  margin: 0;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

input + .c1 {
  margin-left: var(--spacing-half);
}

.c0 {
  margin: 0 0 var(--spacing-3x);
}

.c0 input,
.c0 select,
.c0 textarea {
  border-color: #60666E;
}

.c0 > :nth-child(1) {
  margin-bottom: var(--spacing-half);
}

.c6 {
  background-color: #60666E;
  border: 1px solid #FFFFFF;
  border-radius: var(--border-radius);
  box-shadow: 0 10px 20px 0 rgb(0 0 0 / 19%);
  box-sizing: border-box;
  color: #FFFFFF;
  font-size: 0.75rem;
  padding: var(--spacing-half) var(--spacing-1x);
  position: absolute;
  top: 2.5rem;
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  -webkit-transform-origin: 50% -0.49rem;
  -ms-transform-origin: 50% -0.49rem;
  transform-origin: 50% -0.49rem;
  -webkit-transition: -webkit-transform 0.12s;
  -webkit-transition: transform 0.12s;
  transition: transform 0.12s;
}

.c6::before {
  border-color: transparent transparent #60666E transparent;
  border-style: solid;
  border-width: 0 0.5rem 0.5rem;
  content: '';
  height: 0;
  left: 50%;
  position: absolute;
  top: -0.49rem;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  width: 0;
}

.c7 {
  color: #60666E;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 0.75rem;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-top: 2px;
}

.c3 {
  padding: 0 12px;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 8px;
  padding: 8px 0;
  position: relative;
  -webkit-tap-highlight-color: transparent;
  touch-action: none;
  width: 100%;
}

.c4 .base-Slider-rail {
  background-color: #DBDEE1;
  border-radius: 99px;
  display: block;
  height: 8px;
  margin-left: -12px;
  padding: 0 12px;
  position: absolute;
  width: 100%;
}

.c4 .base-Slider-track {
  background-color: #006296;
  border-radius: 99px;
  display: block;
  height: 8px;
  margin-left: -12px;
  padding: 0 12px;
  position: absolute;
}

.c4 .base-Slider-thumb {
  border-radius: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 32px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-left: -16px;
  position: absolute;
  width: 32px;
}

.c4 .base-Slider-thumb::before {
  background-color: #FFFFFF;
  border: 2px solid #006296;
  border-radius: 50%;
  box-sizing: border-box;
  content: '';
  display: block;
  height: 24px;
  left: 50%;
  outline: 0;
  position: absolute;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  -webkit-transition-duration: 120ms;
  transition-duration: 120ms;
  -webkit-transition-property: box-shadow,width,height;
  transition-property: box-shadow,width,height;
  width: 24px;
}

.c4 .base-Slider-thumb.base--active::before,
.c4 .base-Slider-thumb:hover::before {
  height: 32px;
  width: 32px;
}

.c4 .base-Slider-thumb.base--focusVisible::before {
  box-shadow: inset 0 0 0 2px #84C6EA;
}

.c4 .base-Slider-thumb.base--active::before {
  box-shadow: inset 0 0 0 2px #006296;
}

.c4 .base-Slider-thumb.base--focusVisible .c5,
.c4 .base-Slider-thumb:hover .c5 {
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.c4 .base-Slider-mark {
  background-color: #60666E;
  border-radius: 99%;
  height: 2px;
  position: absolute;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  width: 2px;
}

.c4 .base-Slider-markActive {
  background-color: #FFFFFF;
}

.c4.base--disabled {
  cursor: default;
  pointer-events: none;
}

.c4.base--disabled .base-Slider-rail {
  background-color: #F1F2F2;
}

.c4.base--disabled .base-Slider-track {
  background-color: #B7BBC2;
}

.c4.base--disabled .base-Slider-thumb::before {
  border-color: #B7BBC2;
}

.c4.base--disabled .base-Slider-mark {
  background-color: #B7BBC2;
}

.c4.base--disabled .base-Slider-markActive {
  background-color: #FFFFFF;
}

<div
  class="c0"
>
  <label
    class="c1 c2"
    for="uuid1"
    id="uuid1_label"
  >
    Label
  </label>
  <div
    class="c3"
  >
    <span
      class="base-Slider-root c4"
      data-testid="slider"
    >
      <span
        class="base-Slider-rail"
      />
      <span
        class="base-Slider-track"
        style="left:0%;width:20%"
      />
      <span
        class="base-Slider-thumb"
        data-index="0"
        style="left:20%"
      >
        <input
          aria-label="Label"
          aria-orientation="horizontal"
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="20"
          data-index="0"
          max="100"
          min="0"
          step="1"
          style="border:0;clip:rect(0 0 0 0);height:100%;margin:-1px;overflow:hidden;padding:0;position:absolute;white-space:nowrap;width:100%;direction:ltr"
          type="range"
          value="20"
        />
        <div
          class="c5 c6"
        >
          20
        </div>
      </span>
    </span>
  </div>
  <div
    class="c7"
  >
    <span>
      0
    </span>
    <span>
      100
    </span>
  </div>
</div>
`;

exports[`Slider matches snapshot (with step) 1`] = `
.c2 {
  color: #1B1C1E;
  display: block;
  font-size: 0.75rem;
  font-weight: var(--font-normal);
  -webkit-letter-spacing: 0.02rem;
  -moz-letter-spacing: 0.02rem;
  -ms-letter-spacing: 0.02rem;
  letter-spacing: 0.02rem;
  line-height: 1.25rem;
  margin: 0;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

input + .c1 {
  margin-left: var(--spacing-half);
}

.c0 {
  margin: 0 0 var(--spacing-3x);
}

.c0 input,
.c0 select,
.c0 textarea {
  border-color: #60666E;
}

.c0 > :nth-child(1) {
  margin-bottom: var(--spacing-half);
}

.c6 {
  background-color: #60666E;
  border: 1px solid #FFFFFF;
  border-radius: var(--border-radius);
  box-shadow: 0 10px 20px 0 rgb(0 0 0 / 19%);
  box-sizing: border-box;
  color: #FFFFFF;
  font-size: 0.75rem;
  padding: var(--spacing-half) var(--spacing-1x);
  position: absolute;
  top: 2.5rem;
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  -webkit-transform-origin: 50% -0.49rem;
  -ms-transform-origin: 50% -0.49rem;
  transform-origin: 50% -0.49rem;
  -webkit-transition: -webkit-transform 0.12s;
  -webkit-transition: transform 0.12s;
  transition: transform 0.12s;
}

.c6::before {
  border-color: transparent transparent #60666E transparent;
  border-style: solid;
  border-width: 0 0.5rem 0.5rem;
  content: '';
  height: 0;
  left: 50%;
  position: absolute;
  top: -0.49rem;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  width: 0;
}

.c7 {
  color: #60666E;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 0.75rem;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-top: 2px;
}

.c3 {
  padding: 0 12px;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 8px;
  padding: 8px 0;
  position: relative;
  -webkit-tap-highlight-color: transparent;
  touch-action: none;
  width: 100%;
}

.c4 .base-Slider-rail {
  background-color: #DBDEE1;
  border-radius: 99px;
  display: block;
  height: 8px;
  margin-left: -12px;
  padding: 0 12px;
  position: absolute;
  width: 100%;
}

.c4 .base-Slider-track {
  background-color: #006296;
  border-radius: 99px;
  display: block;
  height: 8px;
  margin-left: -12px;
  padding: 0 12px;
  position: absolute;
}

.c4 .base-Slider-thumb {
  border-radius: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 32px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-left: -16px;
  position: absolute;
  width: 32px;
}

.c4 .base-Slider-thumb::before {
  background-color: #FFFFFF;
  border: 2px solid #006296;
  border-radius: 50%;
  box-sizing: border-box;
  content: '';
  display: block;
  height: 24px;
  left: 50%;
  outline: 0;
  position: absolute;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  -webkit-transition-duration: 120ms;
  transition-duration: 120ms;
  -webkit-transition-property: box-shadow,width,height;
  transition-property: box-shadow,width,height;
  width: 24px;
}

.c4 .base-Slider-thumb.base--active::before,
.c4 .base-Slider-thumb:hover::before {
  height: 32px;
  width: 32px;
}

.c4 .base-Slider-thumb.base--focusVisible::before {
  box-shadow: inset 0 0 0 2px #84C6EA;
}

.c4 .base-Slider-thumb.base--active::before {
  box-shadow: inset 0 0 0 2px #006296;
}

.c4 .base-Slider-thumb.base--focusVisible .c5,
.c4 .base-Slider-thumb:hover .c5 {
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.c4 .base-Slider-mark {
  background-color: #60666E;
  border-radius: 99%;
  height: 2px;
  position: absolute;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  width: 2px;
}

.c4 .base-Slider-markActive {
  background-color: #FFFFFF;
}

.c4.base--disabled {
  cursor: default;
  pointer-events: none;
}

.c4.base--disabled .base-Slider-rail {
  background-color: #F1F2F2;
}

.c4.base--disabled .base-Slider-track {
  background-color: #B7BBC2;
}

.c4.base--disabled .base-Slider-thumb::before {
  border-color: #B7BBC2;
}

.c4.base--disabled .base-Slider-mark {
  background-color: #B7BBC2;
}

.c4.base--disabled .base-Slider-markActive {
  background-color: #FFFFFF;
}

<div
  class="c0"
>
  <label
    class="c1 c2"
    for="uuid1"
    id="uuid1_label"
  >
    Label
  </label>
  <div
    class="c3"
  >
    <span
      class="base-Slider-root c4"
      data-testid="slider"
    >
      <span
        class="base-Slider-rail"
      />
      <span
        class="base-Slider-track"
        style="left:0%;width:30%"
      />
      <span
        class="base-Slider-mark base-Slider-markActive"
        data-index="0"
        style="left:0%"
      />
      <span
        class="base-Slider-mark base-Slider-markActive"
        data-index="1"
        style="left:10%"
      />
      <span
        class="base-Slider-mark base-Slider-markActive"
        data-index="2"
        style="left:20%"
      />
      <span
        class="base-Slider-mark base-Slider-markActive"
        data-index="3"
        style="left:30%"
      />
      <span
        class="base-Slider-mark"
        data-index="4"
        style="left:40%"
      />
      <span
        class="base-Slider-mark"
        data-index="5"
        style="left:50%"
      />
      <span
        class="base-Slider-mark"
        data-index="6"
        style="left:60%"
      />
      <span
        class="base-Slider-mark"
        data-index="7"
        style="left:70%"
      />
      <span
        class="base-Slider-mark"
        data-index="8"
        style="left:80%"
      />
      <span
        class="base-Slider-mark"
        data-index="9"
        style="left:90%"
      />
      <span
        class="base-Slider-mark"
        data-index="10"
        style="left:100%"
      />
      <span
        class="base-Slider-thumb"
        data-index="0"
        style="left:30%"
      >
        <input
          aria-label="Label"
          aria-orientation="horizontal"
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="30"
          data-index="0"
          max="100"
          min="0"
          step="10"
          style="border:0;clip:rect(0 0 0 0);height:100%;margin:-1px;overflow:hidden;padding:0;position:absolute;white-space:nowrap;width:100%;direction:ltr"
          type="range"
          value="30"
        />
        <div
          class="c5 c6"
        >
          30
        </div>
      </span>
    </span>
  </div>
  <div
    class="c7"
  >
    <span>
      0
    </span>
    <span>
      100
    </span>
  </div>
</div>
`;
